.github/workflows/cleanup-azure-resources.yml

```diff
@@ -1,37 +1,120 @@
+name: CleanupAzureResources
+
 on:
   pull_request:
     types: [closed]
 
-name: CleanupAzureResources
+permissions:
+   id-token: write
+   contents: write
 
 jobs:
   cleanup:
-    runs-on: ubuntu-latest
+    runs-on: windows-latest # AzureAD PS module requires Windows.
+
     steps:
-      - name: Login via Az module
+      - name: Checkout Current Branch to import custom Powershell Modules
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+          ref: 'main'
+
+      - name: Azure Login
         uses: azure/login@v1.4.1
         with:
-          creds: ${{ secrets.AZURE_CREDENTIALS }}
+          client-id: '1e8ef5b4-c0cb-428a-a9b4-fc532567bb9d' #${{ secrets.AZ_DEVOPS_APP_ID }}
+          tenant-id: '72f988bf-86f1-41af-91ab-2d7cd011db47' #${{ secrets.AZ_DEVOPS_TENANT_ID }}
+          subscription-id: 'a1766500-6fd5-4f5c-8515-607798271014' #${{ secrets.AZ_DEVOPS_SUBSCRIPTION_ID }}
```

Comment:
Revert this??

Reply:
Done. I was experimenting with the app registration changes.